x/subscription/keeper/subscription.go

```diff
@@ -263,11 +238,44 @@ func (k Keeper) upgradeSubscriptionPlan(ctx sdk.Context, duration uint64, sub *t
 	// Remove one refcount for previous plan
 	k.plansKeeper.PutPlan(ctx, sub.PlanIndex, sub.PlanBlock)
 
-	sub.PlanIndex = toPlan.Index
-	sub.PlanBlock = toPlan.Block
-	sub.MonthCuTotal = toPlan.PlanPolicy.TotalCuLimit
+	sub.PlanIndex = newPlan.Index
+	sub.PlanBlock = newPlan.Block
+	sub.MonthCuTotal = newPlan.PlanPolicy.TotalCuLimit
+
+	k.resetSubscriptionDetailsAndAppendEntry(ctx, sub, nextEpoch, true)
```

Comment:
inside the function we delete the timer if we have error, instead, create the timer only after the append is done